Library/Homebrew/diagnostic.rb

```diff
@@ -345,6 +345,23 @@ def check_access_prefix_directories
         EOS
       end
 
+      def check_missing_prefix_directories
```

Comment:
`return if HOMEBREW_PREFIX.to_s != "/usr/local"` as as-is this breaks CI and isn't really needed for the non-`/usr/local` case (as `sudo` wouldn't be needed there).

Reply:
Sorry, I don't understand this comment. Maybe pasted on the wrong PR?

The code I added checks that the following paths exist:

````
#<Pathname:/usr/local/bin>
#<Pathname:/usr/local/etc>
#<Pathname:/usr/local/include>
#<Pathname:/usr/local/lib>
#<Pathname:/usr/local/sbin>
#<Pathname:/usr/local/share>
#<Pathname:/usr/local/var>
#<Pathname:/usr/local/Frameworks>
#<Pathname:/usr/local/lib/pkgconfig>
#<Pathname:/usr/local/share/locale>
#<Pathname:/usr/local/share/man>
#<Pathname:/usr/local/opt>
````

Reply:
> The code I added checks that the following paths exist:

It doesn't do that because `HOMEBREW_PREFIX` isn't always `/usr/local`. Please add `return if HOMEBREW_PREFIX.to_s != "/usr/local"` as the first line of this function.